akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -147,7 +192,24 @@ private[http2] abstract class Http2Demux(http2Settings: Http2CommonSettings, ini
 
       override protected def logSource: Class[_] = if (isServer) classOf[Http2ServerDemux] else classOf[Http2ClientDemux]
 
-      val multiplexer = createMultiplexer(frameOut, StreamPrioritizer.first())
+      def pushFrameOut(event: FrameEvent): Unit = {
+        pingState match {
+          case OptionVal.None =>
+          case OptionVal.Some(state) =>
+            event match {
+              case _: StreamFrameEvent => state.onDataFrameSeen()
+              case _                   =>
+            }
+        }
+        frameOut.push(event)
+      }
+
+      val multiplexer = createMultiplexer(pushFrameOut, StreamPrioritizer.first())
+      frameOut.setHandler(multiplexer)
```

Comment:
Hmm, this got a bit weird now, but I guess, I don't have any better idea...

Reply:
Why do you think it's weird? My first impression was it looked cleaner. 🤔

Reply:
Because multiplexer before "owned" the outlet but now it's more complex...

Reply:
"What's the relation with the outlet and the multiplexer?"
"It's complicated."